EIPS/eip-2535.md

```diff
@@ -0,0 +1,468 @@
+
+**Who**
+Here are some ways **who** can be limited:
+1. Only allow a trusted individual or organization to make diamond upgrades.
+2. Only allow a distributed autonomous organization to make diamond upgrades.
+3. Only allow multi-signature upgrades.
+4. Only allow the end user who owns his own diamond to make upgrades. This enables users to opt-in to upgrades.
+5. Don't allow anybody to make upgrades by making a single cut diamond.
+
+**When**
+Here are some ways **when** can be limited:
+1. Only allow upgrades during development and testing. Make a single cut diamond for main network release.
+2. Use an upgradeable diamond until it is certain that no new features are needed and then make it a finished diamond by removing the ability to add/replace/remove functions.
+3. Program into the `diamondCut` function certain periods of time that the diamond can be upgraded. For example the `diamondCut` function could be programmed so that a diamond could only be upgraded during a specific five hour period each year. Attention and transparency could be applied to that five hour period to ensure upgrades are done right.
+
+**Transparency**
+Transparency provides certainty and proof that upgrades are done correctly and honestly.
+1. Publish and make available verified source code used by diamonds and facets.
+2. Provide documentation for diamonds, facets, upgrade plans, and results of upgrades.
+3. Provide tools that make your diamonds more visible and understandable.
+
+### Function Selector Clash
+A function selector clash occurs when two different function signatures hash to the same four-byte hash. This has the unintended consequence of replacing an existing function in a diamond when the intention was to add a new function. Function selector clashes are very rare but should be prevented by using tools that check for function selector clashes before calling the `diamondCut` function.
+
+### Transparency
+
+Diamonds emit an event every time one or more functions are added, replaced or removed. All source code can be verified. This enables people and software to monitor changes to a contract. If any bad acting function is added to a diamond then it can be seen. 
+
+Security and domain experts can review the history of change of a diamond to detect any history of foul play.
+
+## Backwards Compatibility
+
+This standard makes upgradeable diamonds compatible with future standards and functionality because new functions can be added and existing functions can be replaced or removed.
+
+
+
+## Contributors
+
+* Andrew Redden
+* Patrick Gallagher
+* Leo Alt
+* Santiago Palladino
+* William Entriken
+* Gonçalo Sá
```

Comment:
The CI spellchecker is complaining about the spelling of the word `alo` here, which suggests to me that it isn't unicode friendly.  I have pinged some people more knowledgeable than I asking how to bypass that particular check, but I don't know how long until they get back to me.  If you don't want to wait, other options are to use ASCII characters for the name to work around the issue or remove this section entirely.

Reply:
I understand. I removed the section.  Ready for merge.